apiserver/controllers/metrics.go

```diff
@@ -0,0 +1,130 @@
+package controllers
+
+import (
+	"log"
+	"os"
+
+	"garm/auth"
+	"garm/runner"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type GarmCollector struct {
+	healthMetric   *prometheus.Desc
+	instanceMetric *prometheus.Desc
+	runner         *runner.Runner
+}
+
+func NewGarmCollector(r *runner.Runner) *GarmCollector {
+	return &GarmCollector{
+		runner: r,
+		instanceMetric: prometheus.NewDesc(
+			"garm_runner_status",
+			"Status of the runner",
+			[]string{"name", "status", "runner_status", "pool_owner", "pool_type", "pool_id", "hostname", "controller_id"}, nil,
+		),
+		healthMetric: prometheus.NewDesc(
+			"garm_health",
+			"Health of the runner",
+			[]string{"hostname", "controller_id"}, nil,
+		),
+	}
+}
+
+func (c *GarmCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- c.instanceMetric
+	ch <- c.healthMetric
+}
+
+func (c *GarmCollector) Collect(ch chan<- prometheus.Metric) {
+	controllerID, err := c.runner.GetControllerID(auth.GetAdminContext())
+	if err != nil {
+		log.Printf("cannot collect metrics, getting controller ID: %s", err)
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		log.Printf("cannot collect metrics, getting hostname: %s", err)
+	}
+
+	c.CollectInstanceMetric(ch, hostname, controllerID.String())
+	c.CollectHealthMetric(ch, hostname, controllerID.String())
+}
+
+func (c *GarmCollector) CollectHealthMetric(ch chan<- prometheus.Metric, hostname string, controllerID string) {
+	ch <- prometheus.MustNewConstMetric(
```

Comment:
This might panic.

Reply:
.. sorry for the confusion .. I thought I hit 're-request review' once I'm done .. but you're too fast

Reply:
I will let you finish :smile:. Will check back later!